### PR TITLE
Migrate ChallengeDialog story

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
Migrates the ChallengeDialog.story.jsx file

Screen Shot 2022-11-14 at 5 28 59 PM

## Links

https://codedotorg.atlassian.net/browse/PP-186

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/ChallengeDialog.story.jsx [hunk 1]
```diff
@@ -3,158 +3,135 @@ import CodeWritten from './feedback/CodeWritten';
...
+  <div style={wrapperStyle}>
+    <ChallengeDialog {...args} />
+  </div>
+);
+
+// STORIES
+
+export const Starting_Dialog = Template.bind({});
+Starting_Dialog.args = {
+  hideBackdrop: true,
```
Comment (bencodeorg) [inline apps/src/templates/ChallengeDialog.story.jsx]: Maybe out of scope, but could we dedupe some of these props? I think either of the approaches discussed here would work:

https://github.com/code-dot-org/code-dot-org/pull/49129#discussion_r1022236463

File: apps/src/templates/ChallengeDialog.story.jsx [hunk 2]
```diff
@@ -3,158 +3,135 @@ import CodeWritten from './feedback/CodeWritten';
...
+  avatar: '/blockly/media/skins/harvester/static_avatar.png',
+  cancelButtonLabel: 'Skip for now',
+  primaryButtonLabel: "I'm Ready!",
+  text:
+    'Challenge Puzzles are lessons designed to push your skills to a new level.',
+  title: 'Challenge Puzzle!',
+  isIntro: true
+};
+
+export const Starting_Dialog_with_Large_Avatar = Template.bind({});
```
Comment (bencodeorg) [inline apps/src/templates/ChallengeDialog.story.jsx]: Nit: are underscores a norm?

File: apps/src/templates/ChallengeDialog.story.jsx (hunk 1 above)
Comment (sanchitmalhotra126) [reply apps/src/templates/ChallengeDialog.story.jsx]: Good call, I'll pull the common ones in

File: apps/src/templates/ChallengeDialog.story.jsx (hunk 2 above)
Comment (sanchitmalhotra126) [reply apps/src/templates/ChallengeDialog.story.jsx]: Oh hmm, I thought I had seen underscores in an example but looking again I may have been looking at something else. I'll remove to be consistent

**Review by maddiedierker — APPROVED** [review]

**Review by bencodeorg — APPROVED** [review]
Oh dang, thought there were more shared props. LGTM!

**Pull request merged**